src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -403,15 +407,23 @@ class BackspacePrediction implements IPrediction {
 		pos: ICoordinate;
 		oldAttributes: string;
 		oldChar: string;
+		eol: boolean;
 	};
 
+	constructor(private readonly terminal: Terminal) { }
+
 	public apply(_: IBuffer, cursor: Cursor) {
+		// at eol if everything to the right is whitespace (zsh will emit a "clear line" code in this case)
+		// todo: can be optimized if `getTrimmedLength` is exposed from xterm
+		const eol = !cursor.getLine()?.translateToString(undefined, cursor.x).trim();
```

Comment:
FYI I think `trim()` and xterm.js' "trimmed length" do different things, echo'd whitespace would not be trimmed for example. But anyway, shouldn't this be comparing `cursor.x` vs the terminal's `cols`?

Reply:
This is more "end of characters". In zshell if you do a backspace on the last character (in some scenarios) it will emit `\b${CSI}K`. I'm not sure why, but that is what happens, and we want to be able to detect whether that is equivalent to a single-character backspace.

Reply:
Oh ok, how about `isLastChar`? EOL to me means when `cursor.x === cols - 1`

Reply:
Sure, pushed to master in 1ea6c9caff